recipes/generate.py

```diff
@@ -100,54 +99,23 @@ def _setup_model(
 
     def convert_prompt_to_tokens(
         self,
-        prompt: Union[DictConfig, str],
-        chat_format: Optional[ChatFormat],
-        instruct_template: Optional[InstructTemplate],
-    ) -> List[Message]:
+        prompt: str,
+    ) -> List[int]:
         """
-        Either:
-        (1) a raw string is passed as the prompt, in which case we call tokenizer.encode directly, or
-        (2) a DictConfig is passed as the prompt. In this case there are three possibilities:
-            (a) an InstructTemplate is provided. Since instruct templates output a string, we will
-                call tokenizer.encode on the output of the instruct template.
-            (b) a ChatFormat is provided. Since chat formats output a list of messages, we will
-                call tokenizer.tokenize_messages on the output of the chat format.
-            (c) neither an InstructTemplate nor a ChatFormat is provided. In this case we will
-                convert the DictConfig to a list of messages and call tokenizer.tokenize_messages directly.
+        Convert the prompt string to a user message and tokenize using the prompt template
+        defined on the tokenizer.
         """
-
-        # Should only be chat-style prompt or instruct-style prompt
-        if chat_format and instruct_template:
-            raise ValueError(
-                "Cannot pass both chat format and instruct template for generation"
-            )
-
-        # If instruct template is provided, assert that the prompt is a DictConfig
-        # and apply it
-        if instruct_template:
-            if not isinstance(prompt, DictConfig):
-                raise ValueError("Cannot apply instruct template to raw string")
-            instruct_template = _get_component_from_path(instruct_template)
-            prompt = instruct_template.format(prompt)
-
-        # To hit this block, either the raw prompt is a string or an
-        # instruct template has been provided to convert it to a string
-        if isinstance(prompt, str):
-            return self._tokenizer.encode(prompt, add_bos=True, add_eos=False)
-
-        # dict.items() will respect order for Python >= 3.7
-        else:
-            messages = [Message(role=k, content=v) for k, v in prompt.items()]
-            messages += [Message(role="assistant", content="")]
-            if chat_format:
-                chat_format = _get_component_from_path(chat_format)
-                messages = chat_format.format(messages)
-            return self._tokenizer.tokenize_messages(messages)[0]
+        messages = [
```

Comment:
noobq: Is this identical to free-form generation when a prompt template isn't provided?

Reply:
No, this assumes instruct-based finetuned models I believe.